src/components/Charts/Axis/Axis.js

```diff
@@ -91,6 +91,32 @@ export default class Axis extends Component {
   }
 
   componentDidUpdate = (prevProps, prevState) => {
+    if (
+      (!_isEqual(prevProps.visibleSeriesIndices, this.props.visibleSeriesIndices) ||
+        prevProps.stringColumnIndex !== this.props.stringColumnIndex) &&
+      this.state.isAxisSelectorOpen !== prevState.isAxisSelectorOpen
```

Comment:
Would it be better/easier to put this stuff in `onChangeNumberColumnIndices` and `on ChangeStringColumnIndex`? Instead of checking every state update? If you have thought about this and did this intentionally then that's fine. Just making sure

Reply:
Sorry I wasn't aware of those two callback function, I will transfer it to there. Thx for your advice!